src/components/NewTable/NewTable.tsx

```diff
@@ -0,0 +1,273 @@
+import React from 'react';
+import {
+  TableWrapper,
+  TableCell,
+  TableHead,
+  TableBody,
+  TableRow,
+  TableHeader,
+  TablePagination,
+  TablePaginationProps,
+} from './components';
+import {
+  ColumnDef,
+  ColumnResizeMode,
+  flexRender,
+  getCoreRowModel,
+  getPaginationRowModel,
+  Row,
+  useReactTable,
+} from '@tanstack/react-table';
+import { css, cx } from '@emotion/css';
+import { Skeleton } from '@mui/material';
+import { tokens } from '../../theme/tokens';
+
+export type NewTableProps<TData> = {
+  columns: ColumnDef<TData>[];
+  data: TData[];
+  isLoading?: boolean;
+  onClick?: (row: Row<TData>) => void;
+  renderEmpty?: () => React.ReactNode;
+  pagination?: TablePaginationProps;
+};
+
+export const NewTable = <TData extends object>({
+  columns,
+  data,
+  isLoading,
+  onClick,
+  pagination,
+  renderEmpty = () => 'No results',
+}: NewTableProps<TData>) => {
+  const columnResizeMode: ColumnResizeMode = 'onChange';
+
+  const table = useReactTable({
+    data,
+    columns,
+    columnResizeMode,
+    getCoreRowModel: getCoreRowModel(),
+    getPaginationRowModel: !!pagination ? getPaginationRowModel() : undefined,
+  });
+
+  const paginationData = table.getState().pagination;
+
+  if (isLoading) {
+    return (
+      <TableWrapper>
+        <TableHeader>
+          {table.getHeaderGroups().map((headerGroup) => (
+            <TableRow key={headerGroup.id}>
+              {headerGroup.headers.map((header) => (
+                <TableHead
+                  key={header.id}
+                  className={styles.head(header.getSize(), header.column.columnDef.enablePinning)}
+                >
+                  {header.isPlaceholder ? null : flexRender(header.column.columnDef.header, header.getContext())}
+                </TableHead>
+              ))}
+            </TableRow>
+          ))}
+        </TableHeader>
+        <TableBody>
+          {[...new Array(10)].map(() => (
```

Comment:
Wanna make it 10 or passed from props. Wdyt

Reply:
Good question, I had the same thought but saw that from the BE we only return 10 items at a time. So I guess it's ok to leave it as is until the BE changes

Reply:
Okay makes sense. I asked because there was a time network offers were 3 max but it later got increased